### PR TITLE
Schema Rework

### PR DESCRIPTION
**Pull request by BoogieMonster1O1**
Condenses most fields into other objects to allow for expansion since codecs can only have sixteen fields. 
Also renamed `dimensions` to `worlds` as it's more clearer.

### CODE REVIEW

**Comment by AMereBagatelle** [comment]
Debatable that it's clearer to change `dimensions` to `worlds` unless that's what Mojang calls them now or something.  To a mod developer, yes it makes more sense, but from a player standpoint it seems like it might be confusing.

**Comment by BoogieMonster1O1** [comment]
> Debatable that it's clearer to change `dimensions` to `worlds` unless that's what Mojang calls them now or something. To a mod developer, yes it makes more sense, but from a player standpoint it seems like it might be confusing.

The concept of a dimension isn't usually what it is.
a 'dimension' can either be a dimension type, or a world. Plain dimensions don't exist in 1.16 as they did in 1.15. 
also, people will understand. it's not that hard once you've read the docs

**Comment by BoogieMonster1O1** [comment]
Almost complete. 
Need to finish writing docs and tests

**Review by FlashyReese — APPROVED** [review]
Apart from the `dimension` or `world` renaming, LGTM.

File: docs/schema-v2.old.md [hunk 1]
```diff
@@ -0,0 +1,160 @@
+# Skybox specification
```
Comment (AMereBagatelle) [inline docs/schema-v2.old.md]: Considering I never released anything with this schema version, there's no need to have a old version

Comment (BoogieMonster1O1) [reply docs/schema-v2.old.md]: I’ve just kept that for my reference. 
was planning to clean up after I finish the docs

File: src/main/java/io/github/amerebagatelle/fabricskyboxes/SkyboxManager.java [hunk 2]
```diff
@@ -1,16 +1,17 @@
...
-import io.github.amerebagatelle.fabricskyboxes.skyboxes.AbstractSkybox;
-import net.minecraft.client.util.math.MatrixStack;
-
 import java.util.ArrayList;
 import java.util.LinkedList;
 import java.util.List;
 import java.util.Objects;
 import java.util.function.Supplier;
 
+import com.google.common.collect.Lists;
```
Comment (AMereBagatelle) [inline src/main/java/io/github/amerebagatelle/fabricskyboxes/SkyboxManager.java:9]: Accidental cleanup code run?

File: src/main/java/io/github/amerebagatelle/fabricskyboxes/skyboxes/AbstractSkybox.java [hunk 3]
```diff
@@ -319,37 +323,29 @@ public void parseJson(JsonObjectWrapper jsonObjectWrapper) {
...
-                if (high > low) {
-                    this.heightRanges.add(new HeightEntry(low, high));
-                } else {
-                    FabricSkyBoxesClient.getLogger().warn("Skybox " + getType() + " contains invalid height ranges.");
+                this.heightRanges.add(new HeightEntry(low, high));
+            }
+        }
+    }
+
+    private void processIds(JsonElement element, List<Identifier> list) {
```
Comment (AMereBagatelle) [inline src/main/java/io/github/amerebagatelle/fabricskyboxes/skyboxes/AbstractSkybox.java:341]: Umm, maybe I'm misunderstanding here, but wouldn't this just add the element to a copied list, not the actual list?

**Review by AMereBagatelle — CHANGES_REQUESTED** [review]
Other than cleanup things that I assume you are already planning to take care of, here's the things I found

File: src/main/java/io/github/amerebagatelle/fabricskyboxes/skyboxes/AbstractSkybox.java (hunk 3 above)
Comment (BoogieMonster1O1) [reply src/main/java/io/github/amerebagatelle/fabricskyboxes/skyboxes/AbstractSkybox.java:341]: This code was duplicated in some methods. Just decided to make a method for it. 
And no, lists are passed by reference, which allows for stuff like this to happen.

File: src/main/java/io/github/amerebagatelle/fabricskyboxes/SkyboxManager.java (hunk 2 above)
Comment (BoogieMonster1O1) [reply src/main/java/io/github/amerebagatelle/fabricskyboxes/SkyboxManager.java:9]: 🤔

**Comment by BoogieMonster1O1** [comment]
everything works now. i've also added the ability to use multiple namespaces cos why not

**Comment by AMereBagatelle** [comment]
Still not sure on the test skyboxes changes... imo should be reverted back to the testmod as it's just simpler

**Comment by BoogieMonster1O1** [comment]
> Still not sure on the test skyboxes changes... imo should be reverted back to the testmod as it's just simpler

it allows easily enabling and disabling skyboxes without having to move files around, reload classes, and reload resources. and since debugging is slower than running normally, it makes sense. the testmod is still there because I need it for the generator

**Comment by AMereBagatelle** [comment]
I see.  It's fine then.

**Comment by AMereBagatelle** [comment]
If you are done, I will merge the PR.

**Comment by BoogieMonster1O1** [comment]
finished it yep
please review the docs. i feel that there could be some improvements though i dont know where

File: docs/schema-v2.md [hunk 4]
```diff
@@ -45,19 +69,121 @@ Example:-
...
+  ],
+  "axis": [
+    0.5,
+    0.5,
+    0.5
+  ]
+}
+```
+
+### Weather
```
Comment (AMereBagatelle) [inline docs/schema-v2.md:164]: Weather should be part of the conditionals object, no?

File: docs/schema-v2.md [hunk 5]
```diff
@@ -69,92 +195,158 @@ Example:-
...
+|    Name    |     Datatype    |                      Description                     |      Required      | Default value |
+|:----------:|:---------------:|:----------------------------------------------------:|:------------------:|:-------------:|
+| `textures` | Textures object | Specifies the textures to be used for each direction | :white_check_mark: |       -       |
+
+### Animated Square Textured skybox
+Only the `animated-square-textured` use these fields
+|         Name        |          Datatype         |                          Description                         |      Required      | Default value |
+|:-------------------:|:-------------------------:|:------------------------------------------------------------:|:------------------:|:-------------:|
+| `animationTextures` | Array of Textures objects | Specifies the list of textures to be used for each direction | :white_check_mark: |       -       |
+| `fps`               | Floating Point            | Specifies the number of frames to be rendered per second     | :white_check_mark: |       -       |
```
Comment (AMereBagatelle) [inline docs/schema-v2.md:352]: Maybe "Specifies the speed of the animation, in frames per second"? (for description of "fps")

**Review by AMereBagatelle — CHANGES_REQUESTED** [review]
I think these were just missed or I can't read git diffs correctly

File: docs/schema-v2.md (hunk 4 above)
Comment (BoogieMonster1O1) [reply docs/schema-v2.md:164]: It is

Comment (AMereBagatelle) [reply docs/schema-v2.md:164]: Oh, I see now what you did.  Looks good now that I am looking at it as a not-git diff

File: docs/schema-v2.md (hunk 5 above)
Comment (AMereBagatelle) [reply docs/schema-v2.md:352]: On second thought, could be improved later but looks fine for now

**Review by AMereBagatelle — APPROVED** [review]

**Pull request merged**